include/Model/Languages.php

```diff
@@ -0,0 +1,1114 @@
+<?php
+/**
+ * @package Polylang
+ */
+
+namespace WP_Syntex\Polylang\Model;
+
+use PLL_Cache;
+use PLL_Language;
+use PLL_Language_Factory;
+use PLL_Translatable_Objects;
+use WP_Error;
+use WP_Term;
+use WP_Syntex\Polylang\Options\Options;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Model for the languages.
+ *
+ * @since 3.7
+ */
+class Languages {
+	public const TRANSIENT_NAME = 'pll_languages_list';
+	private const CACHE_KEY     = 'languages';
+
+	/**
+	 * Polylang's options.
+	 *
+	 * @var Options
+	 */
+	private $options;
+
+	/**
+	 * Translatable objects registry.
+	 *
+	 * @var PLL_Translatable_Objects
+	 */
+	private $translatable_objects;
+
+	/**
+	 * Internal non persistent cache object.
+	 *
+	 * @var PLL_Cache<mixed>
+	 */
+	private $cache;
+
+	/**
+	 * Flag set to true during the language objects creation.
+	 *
+	 * @var bool
+	 */
+	private $is_creating_list = false;
+
+	/**
+	 * Tells if {@see WP_Syntex\Polylang\Model\Languages::get_list()} can be used.
+	 *
+	 * @var bool
+	 */
+	private $languages_ready = false;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 3.7
+	 *
+	 * @param Options                  $options              Polylang's options.
+	 * @param PLL_Translatable_Objects $translatable_objects Translatable objects registry.
+	 * @param PLL_Cache                $cache                Internal non persistent cache object.
+	 *
+	 * @phpstan-param PLL_Cache<mixed> $cache
+	 */
+	public function __construct( Options $options, PLL_Translatable_Objects $translatable_objects, PLL_Cache $cache ) {
```

Comment:
Would it make sense to have a cache internal to this class instead of passing the existing `PLL_Model`'s cache?

Reply:
The current cache object is currently shared among the main model, languages, post types, and taxonomies models.
I think the one for the languages model can be separated without issues, but `clean_languages_cache()` will probably still need to reset all caches.

Reply:
Done in [c33dc1e](https://github.com/polylang/polylang/pull/1503/commits/c33dc1e28de389a649db39c6beb24b54ebf09ddc). Take a look at the hooks I modified in `PLL_Model`'s constructor.

Reply:
Reverted in 562bb4d5fe74efb868c535e1c80ec9e04a6a3682